rust/xaynet-server/src/state_machine/phases/mod.rs

```diff
@@ -70,60 +72,52 @@ pub trait Handler {
     async fn handle_request(&mut self, req: StateMachineRequest) -> Result<(), RequestError>;
 }
 
-/// I/O interfaces.
-#[cfg_attr(test, derive(Debug))]
-pub struct IO {
+/// A struct that contains the coordinator state and the I/O interfaces that are shared and
+/// accessible by all `PhaseState`s.
+pub struct Shared<C, M>
```

Comment:
Maybe we could derive `Debug`, unless some fields aren't `Debug`?

Reply:
Storage is not `Debug` because `Redis` and `S3` do not provide `Debug` and I thought there was no point having an empty debug for both of them.
But I have added a custom `Debug` that includes `state`, `events` and `request_rx`.